controllers/providers/assistant/gslb.go

```diff
@@ -60,27 +61,47 @@ func NewGslbAssistant(client client.Client, k8gbNamespace, edgeDNSServer string)
 
 // CoreDNSExposedIPs retrieves list of IP's exposed by CoreDNS
 func (r *GslbLoggerAssistant) CoreDNSExposedIPs() ([]string, error) {
+	serviceList := &corev1.ServiceList{}
 	coreDNSService := &corev1.Service{}
-	err := r.client.Get(context.TODO(),
-		types.NamespacedName{Namespace: r.k8gbNamespace, Name: coreDNSExtServiceName}, coreDNSService)
+	sel, err := labels.Parse(coreDNSServiceLabel)
+	if err != nil {
+		log.Error().Err(err).Msg("Badly formed label selector")
+		return nil, err
+	}
+	listOption := &client.ListOptions{
+		LabelSelector: sel,
+		Namespace:     r.k8gbNamespace,
+	}
+
+	err = r.client.List(context.TODO(), serviceList, listOption)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			log.Warn().Msgf("Can't find %s service", coreDNSExtServiceName)
+			log.Warn().Err(err).Msg("Can't find CoreDNS service")
+		}
+	}
+	if len(serviceList.Items) != 1 {
+		log.Info().Msgf("items are not 1 actual %d", len(serviceList.Items))
+		for _, service := range serviceList.Items {
```

Comment:
I think it makes sense to log the discovered CoreDNS services before checking their amount on https://github.com/AbsaOSS/k8gb/pull/453/files#diff-b93d6284f641cd9c665d0167b418250059f2ac5438acf1fe775001c13973a368R82
This way the service will be logged in positive case too (for exactly one service found)